cmd/benchsort/main.go

```diff
@@ -0,0 +1,438 @@
+	if valSize <= 0 {
+		return errors.New("value size must be positive")
+	}
+
+	return nil
+}
+
+func export() error {
+	var (
+		err         error
+		outputBytes []byte
+		outputFile  *os.File
+	)
+
+	fileName := path.Join(tmpDir, "data.out")
+	outputFile, err = os.OpenFile(fileName, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer outputFile.Close()
+
+	outputBytes = encodeMeta(outputBytes, scale, keySize, valSize)
+
+	seed := rand.NewSource(time.Now().UnixNano())
+	r := rand.New(seed)
+
+	for i := 1; i <= scale; i++ {
+		outputBytes, err = encodeRow(outputBytes, nextRow(r, keySize, valSize))
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+
+	_, err = outputFile.Write(outputBytes)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	return nil
+}
+
+func load(ratio int) ([]*comparableRow, error) {
+	var (
+		err error
+		fd  *os.File
+	)
+
+	fileName := path.Join(tmpDir, "data.out")
+	fd, err = os.Open(fileName)
+	if os.IsNotExist(err) {
+		return nil, errors.New("data file (data.out) does not exist")
+	}
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer fd.Close()
+
+	err = decodeMeta(fd)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	cLogf("\tnumber of rows = %d, key size = %d, value size = %d", scale, keySize, valSize)
+
+	var (
+		row  *comparableRow
+		data = make([]*comparableRow, 0, scale)
+	)
+
+	totalRows := int(float64(scale) * (float64(ratio) / 100.0))
+	cLogf("\tload %d rows", totalRows)
+	for i := 1; i <= totalRows; i++ {
+		row, err = decodeRow(fd)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		data = append(data, row)
+	}
+
+	return data, nil
+}
+
+func init() {
+	log.SetLevelByString(logLevel)
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	genCmd.StringVar(&tmpDir, "dir", cwd, "where to store the generated rows")
+	genCmd.IntVar(&keySize, "keySize", 8, "the size of key")
+	genCmd.IntVar(&valSize, "valSize", 8, "the size of value")
+	genCmd.IntVar(&scale, "scale", 100, "how many rows to generate")
+	genCmd.StringVar(&cpuprofile, "cpuprofile", "", "write cpu profile to file")
+
+	runCmd.StringVar(&tmpDir, "dir", cwd, "where to load the generated rows")
+	runCmd.IntVar(&bufSize, "bufSize", 500000, "how many rows held in memory at a time")
+	runCmd.IntVar(&inputRatio, "inputRatio", 100, "input percentage")
+	runCmd.IntVar(&outputRatio, "outputRatio", 100, "output percentage")
+	runCmd.StringVar(&cpuprofile, "cpuprofile", "", "write cpu profile to file")
+}
+
+func main() {
+	flag.Parse()
+
+	if len(os.Args) == 1 {
+		fmt.Println("Usage:\n")
+		fmt.Println("\tbenchsort command [arguments]\n")
+		fmt.Println("The commands are:\n")
+		fmt.Println("\tgen\t", "generate rows")
+		fmt.Println("\trun\t", "run tests")
+		fmt.Println("")
+		fmt.Println("Checkout benchsort/README for more information.")
+		return
+	}
+
+	switch os.Args[1] {
+	case "gen":
+		genCmd.Parse(os.Args[2:])
```

Comment:
How about defining a function that generates data, includes the code in `if genCmd.Parsed()` block.
Keep the same logic together.